data/tutorials/lg_08_error_handling.md

```diff
@@ -129,63 +188,534 @@ Each printer should take care of the exceptions it knows about, returning
+stated before there isn't a `Result.bind_error`. One way to make sense out of
+that absence is to consider its type, it would have to be:
+```ocaml
+val Result.bind_error : ('a, 'e) result -> ('e -> ('a, 'f) result) -> ('a, 'f) result
+```
+We would have:
+* `Result.map_error f (Ok x) = Ok x`
+* And either:
+  - `Result.map_error f (Error e) = Ok y`
+  - `Result.map_error f (Error e) = Error e'`
+
+Which means an error would be turned back into valid data, or changed into
+another error. This is almost like recovering from an error. However, when
+recovery fails, it may be preferable to preserve the initial cause of failure.
+That behaviour can be acheived by defining the following function:
+
+```ocaml
+# let recover f = Result.(fold ~ok:ok ~error:(fun (e : 'e) -> Option.to_result ~none:e (f e)));;
+val recover : ('e -> 'a option) -> ('a, 'e) result -> ('a, 'e) result = <fun>
+```
+
+Although any kind of data can be wrapped as a `result` `Error`, it is
+recommended to use that constructor to carry actual errors, for instance:
 - `exn`, in which case the result type just makes exceptions explicit.
 - `string`, where the error case is a message that indicates what failed.
 - `string Lazy.t`, a more elaborate form of error message that is only evaluated
   if printing is required.
-- some polymorphic variant, with one case per
-  possible error. This is very accurate (each error can be dealt with
-  explicitly and occurs in the type) but the use of polymorphic variants
-  sometimes make error messages hard to read.
+- some polymorphic variant, with one case per possible error. This is very
+  accurate (each error can be dealt with explicitly and occurs in the type) but
+  the use of polymorphic variants sometimes make the code harder to read.
+
+Note that some say the types `result` and `Either.t` are
+[ismorphic](https://en.wikipedia.org/wiki/Isomorphism). Concretely, it means
+it's always possible to replace one by the other, like in a completely neutral
+refactoring. Values of type `result` and `Either.t` can be translated back and
+forth, and appling both translations one after the other, in any order, returns
+to the starting value. Nethertheless, this doesn't mean `result` should be used
+in place of `Either.t`, or vise versa. Naming things matters, as punned by Phil
+Karlton famous quote:
+
+> There are only two hard things in Computer Science: cache invalidation and
+> naming things.
+
+Properly handling errors always makes the code harder to read. Using the right
+tools, data and functions can help. Use them.
+
+## `bind` as a Binary Operator
+
+When `Option.bind` or `Result.bind` are used, they are often aliased into a
+custom binding operator, such as `let*`. However, it is also possible to use
+it as binary operator, which is almost always writen `>>=`. Using `bind` this
+way must be detailed because it is extremly popular in other functional
+programming language, and specially in OCaml's arch-rival _Which Must Not Be
```

Comment:
```suggestion
programming languages, and specially in OCaml's arch-rival _Which Must Not Be
```

Also, shouldn't that "Must" become "Shall" if you really intend to keep that reference?